src/diff_xdiff.c

```diff
@@ -160,7 +160,7 @@ static int git_xdiff_cb(void *priv, mmbuffer_t *bufs, int len)
 		else if (line.origin == GIT_DIFF_LINE_DELETION)
 			line.content_offset = bufs[1].ptr - info->xd_old_data.ptr;
 		else
-			line.content_offset = -1;
+			line.content_offset = GIT_OBJECT_SIZE_MAX;
```

Comment:
This changes semantics too, as we pass these to the callers

Reply:
Sorry, can you give me more information about what you mean?

Reply:
We're populating the `git_diff_line` struct here, which we directly pass to the user via the `data_cb` callback. So previously callers would've seen `content_offset = -1`, now he sees `content_offset = UINT64_MAX` if a line was unchanged. I could imagine people having checks for that

Reply:
Moved this back to a signed int to support this.